src/test/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyThingAttachedToCertificateTest.java

```diff
@@ -0,0 +1,133 @@
+        Thing thing = Thing.of("thing-1");
+        X509Certificate certificate = createTestClientCertificate();
+        String certPem = CertificateHelper.toPem(certificate);
+        Certificate thingCertificate = Certificate.fromPem(certPem);
+        iotAuthClientFake.attachCertificateToThing(thing.getThingName(), certPem);
+        VerifyThingAttachedToCertificateDTO dto =
+                new VerifyThingAttachedToCertificateDTO(thing.getThingName(), thingCertificate.getCertificateId());
+
+        when(mockNetworkState.getConnectionStateFromMqtt()).thenReturn(NetworkState.ConnectionState.NETWORK_UP);
+        when(mockThingRegistry.getThing(thing.getThingName())).thenReturn(thing);
+
+        // positive result
+        assertThat(verifyThingAttachedToCertificate.apply(dto), is(true));
+
+        // negative result
+        iotAuthClientFake.detachCertificateFromThing(thing.getThingName(), certPem);
+        assertThat(verifyThingAttachedToCertificate.apply(dto), is(false));
+
+    }
+
+    @Test
+    void GIVEN_validDtoAndNetworkDown_WHEN_verifyThingAttachedToCertificate_THEN_returnLocalResult() throws Exception {
+        Thing thing = Thing.of("thing-1");
+        X509Certificate certificate = createTestClientCertificate();
+        String certPem = CertificateHelper.toPem(certificate);
+        Certificate thingCertificate = Certificate.fromPem(certPem);
+        thing.attachCertificate(thingCertificate.getCertificateId());
+        VerifyThingAttachedToCertificateDTO dto =
+                new VerifyThingAttachedToCertificateDTO(thing.getThingName(), thingCertificate.getCertificateId());
+
+        when(mockNetworkState.getConnectionStateFromMqtt()).thenReturn(NetworkState.ConnectionState.NETWORK_DOWN);
+        when(mockThingRegistry.getThing(thing.getThingName())).thenReturn(thing);
+
+        // positive result
+        assertThat(verifyThingAttachedToCertificate.apply(dto), is(true));
+
+        // negative result
+        thing.detachCertificate(thingCertificate.getCertificateId());
+        assertThat(verifyThingAttachedToCertificate.apply(dto), is(false));
+    }
+
+    @Test
+    void GIVEN_networkUpButFailedCloudCall_WHEN_verifyThingAttachedToCertificate_THEN_returnLocalResult() throws Exception {
+        Thing thing = Thing.of("thing-1");
+        X509Certificate certificate = createTestClientCertificate();
+        String certPem = CertificateHelper.toPem(certificate);
+        Certificate thingCertificate = Certificate.fromPem(certPem);
+        thing.attachCertificate(thingCertificate.getCertificateId());
+        VerifyThingAttachedToCertificateDTO dto =
+                new VerifyThingAttachedToCertificateDTO(thing.getThingName(), thingCertificate.getCertificateId());
+
+        // set up VerifyThingAttachedToCertificate usecase with mock iot auth client
+        // to be able to throw exceptions during cloud calls
```

Comment:
I'd rather add functionality to the fake instead of using a mock here.

The reason is that we may change the iot auth client in the future, and we will likely forget to update this test since CloudServiceInteractionException is a runtime exception.

But we can do that as a separate PR